cmd/amp/storage.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+// StorageCmd is the main command for attaching storage subcommands.
+var StorageCmd = &cobra.Command{
+	Use:   "kv",
+	Short: "Storage operations",
+	Long:  `Manage storage-related operations.`,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		return AMP.Connect()
+	},
+}
+
+var (
+	// storagePutCmd represents the creation of storage key-value pair
+	storagePutCmd = &cobra.Command{
+		Use:   "put [key] [val]",
+		Short: "Assign specified value with specified key",
+		Long:  `Assign specified value with specified key`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return storagePut(AMP, cmd, args)
+		},
+	}
+	// storageGetCmd represents the retrieval of storage value based on key
+	storageGetCmd = &cobra.Command{
+		Use:   "get [key]",
+		Short: "Retrieve a storage object",
+		Long:  `Retrieve a storage object`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return storageGet(AMP, cmd, args)
+		},
+	}
+	// storageDeleteCmd represents the deletion of storage value based on key
+	storageDeleteCmd = &cobra.Command{
+		Use:     "del [key]",
+		Short:   "Delete a storage object (alias: rm)",
+		Long:    `Delete a storage object (alias: rm)`,
+		Aliases: []string{"rm"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return storageDelete(AMP, cmd, args)
+		},
+	}
+	// storageListCmd represents the list of storage key-value pair
+	storageListCmd = &cobra.Command{
+		Use:   "ls",
+		Short: "List all storage objects",
+		Long:  `List all storage objects`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return storageList(AMP, cmd, args)
+		},
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(StorageCmd)
+	StorageCmd.AddCommand(storagePutCmd)
+	StorageCmd.AddCommand(storageGetCmd)
+	StorageCmd.AddCommand(storageDeleteCmd)
+	StorageCmd.AddCommand(storageListCmd)
+}
+
+// storagePut validates the input command line arguments and creates or updates storage key-value pair
+// by invoking the corresponding rpc/storage method
+func storagePut(amp *client.AMP, cmd *cobra.Command, args []string) (err error) {
+	switch len(args) {
+	case 0:
+		return errors.New("Must specify storage key and storage value")
+	case 1:
+		return errors.New("Must specify storage value")
+	case 2:
+		// OK
+	default:
+		return errors.New("Too many arguments")
+	}
+
+	k := args[0]
+	v := args[1]
+	request := &storage.PutStorage{Key: k, Val: v}
+	client := storage.NewStorageClient(amp.Conn)
+	reply, err := client.Put(context.Background(), request)
+	if err != nil {
+		fmt.Println("Key not found: " + k)
+		return nil
+	}
+	fmt.Println(reply.Val)
+	return nil
+}
+
+// storageGet validates the input command line arguments and retrieves storage key-value pair
+//by invoking the corresponding rpc/storage method
+func storageGet(amp *client.AMP, cmd *cobra.Command, args []string) (err error) {
```

Comment:
For all of  these similar functions, there is no need to use named return values since you are very explicit about what you return (specific errors or nil). You can change `(err error)` to `error`.